core/editor/src/macros.rs

```diff
@@ -0,0 +1,15 @@
+macro_rules! count_args {
```

Comment:
Could you please provide a detailed comment describing exactly what each of these do? (Feel free to use multiple lines of comments if needed.) They are nontrivial to read.

Reply:
T0mstone told me to use `count_args`, I can't understand it very well either.

gen_tools_hash_map generates
```rust
{
    let hash_map = HashMap::with_capacity(/* Number of args passed in, counted using `count_args` macro */);

    hash_map.insert(ToolType::Select, select::Select::default());
    hash_map.insert(ToolType::Crop, crop::Crop::default());

    hash_map
}
```
from
```rust
gen_tools_hash_map! {
    Select => select::Select,
    Crop => crop::Crop,
}
```

Reply:
Oh to clarify, I meant as comments in the code 😊